backend/site-api/src/db/site_repo.rs

```diff
@@ -186,21 +189,23 @@ impl SiteRepoTrait for SiteRepo {
             return Err(DbError::NoUpdate);
         }
 
+        query.push(" WHERE id = (SELECT MAX(id) FROM site_versions)");
         // Update the latest version of the site
-        query.push(
-            " WHERE id = (
-                    SELECT MAX(id)
-                    FROM site_versions
-                )",
-        );
+        if let Some(update_key) = req.update_key {
+            // Append the condition to check the updated_at
+            query.push(" AND updated_at = DATETIME(");
+            query.push_bind(update_key);
```

Comment:
I made a small adjustment here, even though we verify `update_key` in the DTO, it's good to use `push_bind` to avoid the possibility of SQL injection.